dashboard/app/models/lesson.rb

```diff
@@ -227,6 +227,12 @@ def lesson_plan_pdf_url
     end
   end
 
+  def student_lesson_plan_pdf_url
+    if script.is_migrated && script.include_student_lesson_plans && has_lesson_plan && Services::LessonPlanPdfs.student_pdf_exists_for?(self)
```

Comment:
we don't want to have make an S3 request every time this helper method gets called; I recommend just doing
```suggestion
    if script.is_migrated && script.include_student_lesson_plans && has_lesson_plan
```

Reply:
> A bunch of tests are failing like the following example. Any idea whats up?

This is what's up; we have tests that are trying to render content, which calls this method, which calls `student_pdf_exists_for`, which then tries to call S3. If we were going to go with this implementation we'd need to stub out all the S3 calls in our tests, but rather than that we should just remove this.